core/src/main/java/org/apache/sling/testing/mock/osgi/config/annotations/UpdateConfig.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.sling.testing.mock.osgi.config.annotations;
+
+import org.osgi.service.component.annotations.Component;
+
+import java.lang.annotation.Repeatable;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+
+/**
+ * Define this annotation on a test class or method to use the {@link org.osgi.service.cm.ConfigurationAdmin} service
+ * to update the persisted properties for the configuration whose pid matches the {@link #pid()} attribute.
+ * Updates should be applied top-down for each test context scope, from with the outermost (class-level) to the
+ * innermost (method-level).
+ */
+@Repeatable(UpdateConfigs.class)
+@Retention(RetentionPolicy.RUNTIME)
+public @interface UpdateConfig {
```

Comment:
i'm wondering if "SetConfig" would be a more intuitive name? if i read update i always look for a "create" which happened first, or updating something that was already set before.
setting something includes creating new or overwriting existing data.

it's called update in "ConfigurationAdmin", but there you are getting a configuration instance first hand and then update it which makes sense. but that's not the case for the annotation here.

Reply:
renamed it to `@SetConfig`